backend/LexBoxApi/GraphQL/ProjectMutations.cs

```diff
@@ -84,6 +89,60 @@ public record CreateProjectResponse(Guid? Id, CreateProjectResult Result);
         return dbContext.Projects.Where(p => p.Id == input.ProjectId);
     }
 
+    [Error<NotFoundException>]
+    [Error<DbError>]
+    [AdminRequired]
+    [UseMutationConvention]
+    public async Task<BulkAddProjectMembersResult> BulkAddProjectMembers(
+        LoggedInContext loggedInContext,
+        BulkAddProjectMembersInput input,
+        LexBoxDbContext dbContext)
+    {
+        var admin = await dbContext.Users.FindAsync(loggedInContext.User.Id);
+        var project = await dbContext.Projects.FindAsync(input.ProjectId);
+        if (project is null) throw new NotFoundException("Project not found");
+        List<string> usernameConflicts = [];
+        int count = 0;
+        foreach (var username in input.Usernames)
+        {
+            var user = await dbContext.Users.Where(u => u.Username == username).FirstOrDefaultAsync();
+            if (user is null)
+            {
+                count++;
+                var salt = Convert.ToHexString(RandomNumberGenerator.GetBytes(SHA1.HashSizeInBytes));
+                user = new User
+                {
+                    Id = Guid.NewGuid(),
+                    Username = username,
+                    Name = username,
+                    Email = null,
```

Comment:
How hard would it be to check if the username is an email?

Reply:
Good point. I was simply assuming that the usernames would NOT be emails, but that won't always be true. We do plan on also allowing admins to paste something like `Firstname Lastname <email@example.com>, Second Person <other@example.com>` into the box. I don't handle that yet, but we will eventually. So I shouldn't assume so much here.

Reply:
Done in commit 24a59579b0da90eff7420d4b0d460b3ee2185681.